plugins/extrusions/extrusions.py

```diff
@@ -0,0 +1,146 @@
+from typing import TypeVar, Union, Tuple
+
+# import importlib.resources
+import pkg_resources
+import cadquery as cq
+from cadquery.occ_impl import importers
+from cadquery.occ_impl.geom import Vector, Location
+
+T = TypeVar("T", bound="Workplane")
+
+
+def _extrusion(
+    size: Tuple[float, float],
+    profile,
+    length: float,
+    centered: Union[bool, Tuple[bool, bool, bool]],
+):
+    """
+    Return an extrusion with the specified profile and length
+
+    :param size: size of the extrusion, used for centering
+    :param profile: filename of the extrusion's profile
+    :param length: length of the extrusion
+    :type length: float > 0
+    :param centered: If True, the extrusion will be centered around the reference point.
```

Comment:
Did you create the DXF profiles yourself? If so you controlled how they were drawn, and thus (I assume) where ezdxf would set the center. ezdxf could be doing something here I don't expect and ignoring the as-drawn center and using the extents to set the center, but I doubt it. If each drawing can have a different center, can you really assume that this centered setting will work as expected?

Reply:
Nevermind, I read further down in your code and I see how you are using the size.